azext_iot/iothub/_help.py

```diff
@@ -309,3 +309,320 @@ def load_iothub_help():
+                --collection {collection} --db {database}
+          - name: Create a Cosmos DB Collection endpoint for an IoT Hub using the specified primary key and endpoint uri.
+            text: >
+                az iot hub message-endpoint create cosmosdb-collection -n {iothub_name} --en {endpoint_name} --pk {primary_key}
+                --endpoint-uri {endpoint_uri} --collection {collection} --db {database}
+          - name: Create a Cosmos DB Collection endpoint for an IoT Hub using system assigned identity and a partition key name.
+                  The partition key template will be the default.
+            text: >
+                az iot hub message-endpoint create cosmosdb-collection -n {iothub_name} --en {endpoint_name} --endpoint-uri {endpoint_uri}
+                --collection {collection} --db {database} --auth-type identityBased --pkn {partition_key_name}
+          - name: Create a Cosmos DB Collection endpoint for an IoT Hub using user assigned identity, partition key name, and partition key template.
+            text: >
+                az iot hub message-endpoint create cosmosdb-collection -n {iothub_name} --en {endpoint_name} --endpoint-uri {endpoint_uri}
+                --collection {collection} --db {database} --auth-type identityBased --pkn {partition_key_name} --pkt {partition_key_template}
+                --identity {user_identity_resource_id}
+    """
+
+    helps[
+        "iot hub message-endpoint create eventhub"
+    ] = """
+        type: command
+        short-summary: Add an Event Hub endpoint for an IoT Hub.
+        examples:
+          - name: Create a key-based Event Hub endpoint for an IoT Hub.
+            text: >
+                az iot hub message-endpoint create eventhub -n {iothub_name} --en {endpoint_name} --namespace {namespace_name}
+                --entity-path {entity_path} --policy {policy_name}
+          - name: Create an Event Hub endpoint for an IoT Hub using a connection string. The endpoint uri and entity path are omitted.
+            text: >
+                az iot hub message-endpoint create eventhub -n {iothub_name} --en {endpoint_name} -c {connection_string}
+          - name: Create an Event Hub endpoint for an IoT Hub using system assigned identity. The endpoint and entity path must be specified.
+            text: >
+                az iot hub message-endpoint create eventhub -n {iothub_name} --en {endpoint_name} --endpoint-uri {endpoint_uri}
+                --auth-type identityBased --entity-path {entity_path}
+          - name: Create an Event Hub endpoint for an IoT Hub using user assigned identity. The endpoint and entity path must be specified.
+            text: >
+                az iot hub message-endpoint create eventhub -n {iothub_name} --en {endpoint_name} --endpoint-uri {endpoint_uri}
+                --auth-type identityBased --entity-path {entity_path} --identity {user_identity_resource_id}
+    """
+
```

Comment:
@c-ryan-k @digimaun do we not need a --endpoint-type for each type of endpoint anymore? now it is just create cosmosdb-collection, create eventhub?

Reply:
Yea that's right, this new command group is conforming to our digital twins endpoint style where each endpoint has a dedicated create command rather than trying to support all endpoints in one. 

For the second one, each create should support `--endpoint-subscription-id` and `--endpoint-resource-group` (as well as `--connection-string`)

Reply:
yup to both - 

sub id and rg is now optional, defaulting to dt sub and rg if not provided
![image](https://user-images.githubusercontent.com/73560279/207448404-10c2c74e-7a21-491a-a7cb-d62f2de461b8.png)